src/communication/request_client_server.hpp

```diff
@@ -267,6 +267,8 @@ namespace cubcomm
     : m_channel (std::move (chn))
     , m_recv_extensible_block { cubmem::CSTYLE_BLOCK_ALLOCATOR }
   {
+    // arbitrary initial size; will be grown upon need
+    m_recv_extensible_block.extend_to (IO_MAX_PAGE_SIZE * 4);
```

Comment:
Nit.
I'm not sure how `IO_MAX_PAGE_SIZE` is related to network communication. 
If it is intended to consider the exchange of pages between servers, it might be a good idea to define a separate constant for that purpose. 

And that constant could be utilized when resizing the block. (`m_recv_extensible_block.extend_by(constant value)`)

Reply:
As commented, it is an arbitrary value.
Could even be set to 1 and the logic will adapt to the actually required size.